From a887482709311c3336e1f8ddcc0884f7af583572 Mon Sep 17 00:00:00 2001
From: guoweikang <guoweikang.kernel@gmail.com>
Date: Wed, 10 Apr 2024 16:14:05 +0800
Subject: [PATCH] Rust support OSL crate lib

Signed-off-by: guoweikang <guoweikang.kernel@gmail.com>
---
 rust/Makefile          | 7 ++++++-
 scripts/Makefile.build | 2 +-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/rust/Makefile b/rust/Makefile
index af6c3bec3cb5..49084be4aa58 100644
--- a/rust/Makefile
+++ b/rust/Makefile
@@ -15,7 +15,7 @@ always-$(CONFIG_RUST) += libmacros.so
 no-clean-files += libmacros.so
 
 always-$(CONFIG_RUST) += bindings/bindings_generated.rs bindings/bindings_helpers_generated.rs
-obj-$(CONFIG_RUST) += alloc.o bindings.o kernel.o
+obj-$(CONFIG_RUST) += alloc.o bindings.o kernel.o osl.o
 always-$(CONFIG_RUST) += exports_alloc_generated.h exports_bindings_generated.h \
     exports_kernel_generated.h
 
@@ -436,4 +436,9 @@ $(obj)/kernel.o: $(src)/kernel/lib.rs $(obj)/alloc.o $(obj)/build_error.o \
     $(obj)/libmacros.so $(obj)/bindings.o $(obj)/uapi.o FORCE
 	$(call if_changed_dep,rustc_library)
 
+$(obj)/osl.o: private rustc_target_flags = --extern alloc --extern kernel --cfg 'feature="linux"'
+$(obj)/osl.o: $(src)/osl/src/lib.rs $(obj)/alloc.o $(obj)/build_error.o \
+    $(obj)/libmacros.so $(obj)/bindings.o $(obj)/uapi.o $(obj)/kernel.o FORCE
+	$(call if_changed_dep,rustc_library)
+
 endif # CONFIG_RUST
diff --git a/scripts/Makefile.build b/scripts/Makefile.build
index 92c920b6cc37..9eeb9b5b763b 100644
--- a/scripts/Makefile.build
+++ b/scripts/Makefile.build
@@ -285,7 +285,7 @@ rust_common_cmd = \
 	-Zallow-features=$(rust_allowed_features) \
 	-Zcrate-attr=no_std \
 	-Zcrate-attr='feature($(rust_allowed_features))' \
-	--extern alloc --extern kernel \
+	--extern alloc --extern kernel --extern osl \
 	--crate-type rlib -L $(objtree)/rust/ \
 	--crate-name $(basename $(notdir $@)) \
 	--out-dir $(dir $@) --emit=dep-info=$(depfile)
-- 
2.25.1

